lib/hw_id/Kconfig

```diff
@@ -0,0 +1,42 @@
+#
+# Copyright (c) 2022 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+menuconfig HW_ID_LIBRARY
+	bool "Device ID"
+	help
+	  Provides a unified interface for fetching a unique device ID from hardware.
+	select HWINFO
+
+if HW_ID_LIBRARY
+
+choice HW_ID_LIBRARY_SOURCE
+	prompt "Which type of ID to use as HW ID"
+
+config HW_ID_LIBRARY_SOURCE_IMEI
+	bool "Enable usage of IMEI as HW ID"
+	depends on NRF_MODEM_LIB
+
+config HW_ID_LIBRARY_SOURCE_UUID
+	bool "Enable usage of UUID as HW ID"
+	depends on MODEM_JWT
+	help
+	  This option requires modem firmware v1.3.0 or higher.
+
+config HW_ID_LIBRARY_SOURCE_BLE_MAC
+	bool "Enable usage of BLE MAC address as HW ID"
+	depends on BT
+
+config HW_ID_LIBRARY_SOURCE_NET_MAC
+	bool "Enable usage of network MAC address as HW ID"
+	depends on NETWORKING
```

Comment:
Currently, this will fail with an error code because the modem will not give us a network interface. Still, I would not disable this option in build, since there might be configurations with more network interfaces that could support that.